google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java

```diff
@@ -291,7 +292,8 @@ private Builder() {
           .setRetrySettings(longRunningRetrySettings);
       databaseAdminStubSettingsBuilder
           .updateBackupSettings()
-          .setRetrySettings(longRunningRetrySettings);
+          .setRetrySettings(longRunningRetrySettings)
+          .setRetryableCodes(StatusCode.Code.DEADLINE_EXCEEDED, StatusCode.Code.UNAVAILABLE);
```

Comment:
Why do we need to manually specify the retryable codes here? In the gapic config file (https://github.com/googleapis/googleapis/blob/d741cd976975c745d0199987aff0e908b8352992/google/spanner/admin/database/v1/spanner_admin_database_grpc_service_config.json#L58-L67), `updateBackup` will retry DEADLINE_EXCEEDED and UNAVAILABLE. Shouldn't this code be auto-generated? 

I guess Java gapic code generator is using a different config file: https://github.com/googleapis/googleapis/blob/master/google/spanner/admin/database/v1/spanner_admin_database_gapic.yaml#L83-L86.

Reply:
Just curious: why do we manually set retrying settings instead of using the default values from the gapic config file?

Reply:
I had a lot of problems / transient errors using the default values. So I set the values manually to try to find settings that did work, so that we can set them back to the default config.

So the Java client should definitely not continue to use custom retry settings. Once we have verified that these settings work well, we should set these as the default in the config file.

Reply:
And yes, the Java client (and I think also the other clients, but I'm not 100% sure) use the retry settings in https://github.com/googleapis/googleapis/blob/master/google/spanner/admin/database/v1/spanner_admin_database_gapic.yaml. So that means for example that the default config for `UpdateBackup` is [non-idempotent](https://github.com/googleapis/googleapis/blob/master/google/spanner/admin/database/v1/spanner_admin_database_gapic.yaml#L83-L86).